### PR TITLE
to_cpu and to_gpu move LSTM's states

### LINKED ISSUE #818: link.LSTM.to_cpu (resp. to_gpu) does not move state variables to CPU (resp. GPU).
When `link.LSTM` is move to CPU (resp. GPU), its state variables should also be moved to CPU(resp. GPU). But it does not.

```
In [39]: l = chainer.links.LSTM(10, 10)

In [40]: l.xp
Out[40]: <module 'numpy' from '/home/delta/.pyenv/versions/pyenv-2.7.9/lib/python2.7/site-packages/numpy/__init__.pyc'>

In [41]: x = chainer.Variable(numpy.random.uniform(-1, 1, (5, 10)).astype(numpy.float32))

In [42]: l(x)
Out[42]: <variable at 0x7fe900a88d90>

In [43]: type(l.h.data)
Out[43]: numpy.ndarray

In [44]: l.to_gpu()
Out[44]: <chainer.links.connection.lstm.LSTM at 0x7fe900a42a10>

In [45]: l.xp
Out[45]: <module 'cupy' from 'cupy/__init__.pyc'>

In [46]: type(l.h.data)
Out[46]: numpy.ndarray

In [47]: type(l.c.data)
Out[47]: numpy.ndarray
```
The output of [46] and [47] should be `cupy.ndarray`

### PR DESCRIPTION
**Pull request by delta2323**
Fix #818

### CODE REVIEW

File: tests/chainer_tests/links_tests/connection_tests/test_lstm.py [hunk 1]
```diff
@@ -57,4 +57,94 @@ def test_forward_gpu(self):
...
+
+    def setUp(self):
+        self.link = links.LSTM(5, 7)
+        self.x = chainer.Variable(
+            numpy.random.uniform(-1, 1, (3, 5)).astype(numpy.float32))
+
+    def check_to_cpu(self, s):
+        self.link.to_cpu()
+        self.assertIsInstance(s.data, self.link.xp.ndarray)
+        self.link.to_cpu()
```
Comment (okuta) [inline tests/chainer_tests/links_tests/connection_tests/test_lstm.py]: Why do you check twice?

Comment (delta2323) [inline tests/chainer_tests/links_tests/connection_tests/test_lstm.py]: First assert tries to check that `to_cpu` forcibly moves the state to CPU even if link is on CPU or GPU. Second assert tries to check the idempotency of `to_cpu`.

Comment (okuta) [inline tests/chainer_tests/links_tests/connection_tests/test_lstm.py]: OK. I understand.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**